### PR TITLE
Clean/tests

### PR DESCRIPTION
**Pull request by OmgImAlexis**
I've changed over all vars to either let or const, there may still be a few vars but they're likely there because of some weird var hoisting rule since we have some functions that are called within themselves.

The `function()` calls were replaced with fat arrows.

The top level describe blocks were removed from the last test PR I merged so now each block is only called by it's filename.

### CODE REVIEW

**Comment by OmgImAlexis** [comment]
@simison would you mind reviewing this as well please.

File: test/fixtures/agenda-instance.js [hunk 1]
```diff
@@ -1,28 +1,30 @@
...
-var tests = process.argv.slice(3);
-
-var path = require('path'),
-    Agenda = require( path.join(__dirname, '..', '..', 'index.js' ) ),
-    addTests = require( path.join(__dirname, 'addTests.js') );
-
-var agenda = new Agenda({ db: { address: connStr } }, function(err, collection) {
-
-	tests.forEach(function(test) {
+const Agenda = require('../../index');
```
Comment (OmgImAlexis) [inline test/fixtures/agenda-instance.js]: The reason I switched to this is if we want to move to ES6 with babel in the future to allow for tree shaking we would likely end up using `import x from '../x';` and you can't use dynamic file paths with imports. Also it's for the most part considered a bad idea to use dynamic requires like that.

Maybe we should open an issue for this discussion so we can add an eslint plugin to force the chosen way?

Comment (simison) [inline test/fixtures/agenda-instance.js]: Aight, valid point! Seems like good to me like it is now as a transition towards XO — how was it, _Never discuss code style on a pull request again!_ :-D

**Comment by simison** [comment]
I skimmed trough and looks good, but unfortunately I'm short on time today for more in depth look. I'll have more changes over the weekend!

File: test/retry.js [hunk 2]
```diff
@@ -52,43 +43,41 @@ describe('agenda', function() {
...
 
-      jobs.now('a job');
+    jobs.now('a job');
 
-      jobs.start();
-    });
+    jobs.start();
   });
-
 });
```
Comment (simison) [inline test/retry.js]: Doesn't XO add extra empty line here?

File: test/retry.js [hunk 3]
```diff
@@ -52,43 +43,41 @@ describe('agenda', function() {
...
+    });
 
-      jobs.on('fail:a job', function (err, job) {
-        job
-          .schedule('now')
-          .save();
-      });
+    jobs.on('fail:a job', (err, job) => {
+      if (err) {
+        // Do nothing as this is expected to fail.
```
Comment (simison) [inline test/retry.js]: Not in scope for this PR, but as a food for thought for future, we should test this kind of stuff with:
```js
expect(err).not.to.be(undefined);
```

(Tho Expect.js probably has some better ways to test error objects)

And that said, we should probably move away from Expect.js as it wasn't maintained in ages anymore.

Comment (OmgImAlexis) [inline test/retry.js]: My thoughts exactly.

I'll switch from mocha + expect.js + blanket to [avajs](https://github.com/avajs/ava) which will get us using the latest testing and coverage setup. It has something exactly for this.

File: test/retry.js (hunk 2 above)
Comment (OmgImAlexis) [inline test/retry.js]: Yep but the current `.editorconfig` removes it. I'll add them back once I start on the large xo changes.

**Comment by lushc** [comment]
LGTM, very much looking forward to those xo and ava changes!

**Pull request merged**